google-cloud-storage/src/main/java/com/google/cloud/storage/Notification.java

```diff
@@ -0,0 +1,320 @@
+
+  public enum PayloadFormat {
+    JSON_API_V1,
+    NONE
+  }
+
+  static final Function<com.google.api.services.storage.model.Notification, Notification>
+      FROM_PB_FUNCTION =
+          new Function<com.google.api.services.storage.model.Notification, Notification>() {
+            @Override
+            public Notification apply(com.google.api.services.storage.model.Notification pb) {
+              return Notification.fromPb(pb);
+            }
+          };
+  static final Function<Notification, com.google.api.services.storage.model.Notification>
+      TO_PB_FUNCTION =
+          new Function<Notification, com.google.api.services.storage.model.Notification>() {
+            @Override
+            public com.google.api.services.storage.model.Notification apply(
+                Notification notification) {
+              return notification.toPb();
+            }
+          };
+  private final String generatedId;
+  private final String topic;
+  private final List<String> eventTypes;
+  private final Map<String, String> customAttributes;
+  private final PayloadFormat payloadFormat;
+  private final String objectNamePrefix;
+  private final String etag;
+  private final String selfLink;
+
+  /** Builder for {@code Notification}. */
+  public static class Builder {
+
+    private String generatedId;
+    private String topic;
+    private List<String> eventTypes;
+    private Map<String, String> customAttributes;
+    private PayloadFormat payloadFormat;
+    private String objectNamePrefix;
+    private String etag;
+    private String selfLink;
+
+    private Builder(String topic) {
+      this.topic = topic;
+    }
+
+    private Builder(Notification notification) {
+      generatedId = notification.generatedId;
+      etag = notification.etag;
+      selfLink = notification.selfLink;
+      topic = notification.topic;
+      eventTypes = notification.eventTypes;
+      customAttributes = notification.customAttributes;
+      payloadFormat = notification.payloadFormat;
+      objectNamePrefix = notification.objectNamePrefix;
+    }
+
+    public Builder setGeneratedId(String generatedId) {
+      this.generatedId = generatedId;
+      return this;
+    }
+
+    public Builder setSelfLink(String selfLink) {
+      this.selfLink = selfLink;
+      return this;
+    }
+
+    /** Sets a topic in the format of "projects/{project}/topics/{topic}". */
+    public Builder setTopic(String topic) {
+      this.topic = topic;
+      return this;
+    }
+
+    public Builder setPayloadFormat(PayloadFormat payloadFormat) {
+      this.payloadFormat = payloadFormat;
+      return this;
+    }
+
+    public Builder setObjectNamePrefix(String objectNamePrefix) {
+      this.objectNamePrefix = objectNamePrefix;
+      return this;
+    }
+
+    public Builder setEventTypes(Iterable<String> eventTypes) {
+      this.eventTypes = eventTypes != null ? ImmutableList.copyOf(eventTypes) : null;
+      return this;
+    }
```

Comment:
Here we cannot do that because `setEventTypes `requires the list parameter you can see  [Notification#setEventTypes](https://developers.google.com/resources/api-libraries/documentation/storage/v1/java/latest/com/google/api/services/storage/model/Notification.html#setEventTypes-java.util.List-) of underlying Google API client object model.

Reply:
`EventType...`  is an array of EventType. You can covert it to the list of String inside of the method. It's much convenient for the user to call `setEventTypes(TYPE1, TYPE2)` rather than create Iterable<String>. Because the set of values is limited, please use an enum to list all the valid values.

Reply:
done